File: Golang/1922-count-good-numbers.go
package main

const mod = 1e9 + 7

func countGoodNumbers(n int64) int {
	return int((fastExp(5, (n+1)/2) * fastExp(4, n/2)) % mod)
}

func fastExp(base, exponent int64) int {
	res := 1
	b := int(base)
	e := exponent
	b %= mod

	for e > 0 {
		if e%2 == 1 {
			res = (res * b) % mod
		}
		b = (b * b) % mod
		e /= 2
	}

	return res
}

File: TS/1922-count-good-numbers.ts
function countGoodNumbers(n: number): number {
  const modulo = 1e9 + 7;

  return (
    (fastExp(5, Math.floor((n + 1) / 2), modulo) *
      fastExp(4, Math.floor(n / 2), modulo)) %
    modulo
  );
}

function fastExp(base: number, exponent: number, mod: number): number {
  let res = 1;
  base = base % mod;

  while (exponent > 0) {
    if (exponent % 2 === 1) {
      res = (res * base) % mod;
    }
    base = (base * base) % mod;
    exponent = Math.floor(exponent / 2);
  }

  return res;
}

console.log(countGoodNumbers(1924));
